python_modules/dagster/dagster/_core/remote_representation/external.py

```diff
@@ -518,6 +523,28 @@ def __init__(
             repository_handle=repository_handle,
         )
 
+    @classmethod
+    async def _batch_load(
+        cls, keys: Iterable[JobSubsetSelector], context: "LoadingContext"
+    ) -> Iterable[Optional["RemoteJob"]]:
+        from dagster._core.workspace.context import BaseWorkspaceRequestContext
+
+        check.invariant(isinstance(context, BaseWorkspaceRequestContext))
+        request_context = cast("BaseWorkspaceRequestContext", context)
```

Comment:
when i first added the `LoadableBy` class i thought it might have a few different flavors for what argument it operated on 

might be able to make it so `LoadableBy[Tkey, TContext]` works 

nit: can do `request_context = check.inst(context, BaseWorkspaceRequestContext)` to combine cast and check